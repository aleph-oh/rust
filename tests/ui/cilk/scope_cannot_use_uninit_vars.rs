#![feature(cilk)]
// Tests that scope can't return a value resulting from uninitialized variables.

// compile-flags: -C panic=abort
// no-prefer-dynamic

fn fib(n: u8) -> usize {
    if n <= 1 {
        return n as usize;
    }

    cilk_scope {
        let x = cilk_spawn { fib(n - 1) };
        let y = fib(n - 2);
        x + y
    //~^ ERROR used binding `x` is possibly-uninitialized [E0381]
    }
}

fn main() {
    assert_eq!(fib(10), 55);
}
