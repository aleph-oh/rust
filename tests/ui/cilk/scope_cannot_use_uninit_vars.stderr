error[E0381]: used binding `x` is possibly-uninitialized
  --> $DIR/scope_cannot_use_uninit_vars.rs:15:9
   |
LL |         let x = cilk_spawn { fib(n - 1) };
   |             -              -------------- binding initialized here in some conditions
   |             |
   |             binding declared here but left uninitialized
LL |         let y = fib(n - 2);
LL |         x + y
   |         ^ `x` used here but it is possibly-uninitialized

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0381`.
